src/main/java/core/basesyntax/utils/generate/ReportGenerator.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.utils.generate;
+
+import core.basesyntax.storage.impl.StorageImpl;
+
+public interface ReportGenerator {
+    String getReport(StorageImpl storage);
```

Comment:
According to the SOLID principles, the `ReportGenerator` interface should not depend directly on the `StorageImpl` class. Instead, it should depend on a `Storage` interface. This will allow for more flexibility and easier testing, as you can provide different implementations of `Storage`.